src/torznab.ts

```diff
@@ -681,3 +642,98 @@ async function makeRequests(
 		candidates: results,
 	}));
 }
+async function getAndLogIndexers(
+	searchee: Searchee,
+	mediaType: MediaType,
+	progress: string,
+): Promise<{ indexersToUse: Indexer[]; parsedMedia?: ParsedMedia }> {
+	const { excludeRecentSearch, excludeOlder } = getRuntimeConfig();
+	const searcheeLog = getLogString(searchee, chalk.bold.white);
+	const mediaTypeLog = chalk.white(mediaType.toUpperCase());
+
+	const enabledIndexers = await getEnabledIndexers();
+
+	// search history for name across all indexers
+	const name = searchee.name;
+	const timestampDataSql = await db("searchee")
+		.join("timestamp", "searchee.id", "timestamp.searchee_id")
+		.join("indexer", "timestamp.indexer_id", "indexer.id")
+		.whereIn(
+			"indexer.id",
+			enabledIndexers.map((i) => i.id),
+		)
+		.andWhere({ name })
+		.select({
+			indexerId: "indexer.id",
+			firstSearched: "timestamp.first_searched",
+			lastSearched: "timestamp.last_searched",
+		});
+
+	const skipBefore = excludeOlder
+		? nMsAgo(excludeOlder)
+		: Number.NEGATIVE_INFINITY;
+	const skipAfter = excludeRecentSearch
+		? nMsAgo(excludeRecentSearch)
+		: Number.POSITIVE_INFINITY;
+	const indexersToUseByTimestamp = enabledIndexers.filter((indexer) => {
+		const entry = timestampDataSql.find(
+			(entry) => entry.indexerId === indexer.id,
+		);
+		if (!entry) return true;
+		if (entry.firstSearched < skipBefore) return false;
+		if (entry.lastSearched > skipAfter) return false;
+		return true;
+	});
+
+	const indexersToUse = indexersToUseByTimestamp.filter((indexer) => {
+		return indexerDoesSupportMediaType(
+			mediaType,
+			JSON.parse(indexer.categories),
+		);
+	});
+	const skippingSearching = (reason: string) =>
+		`${progress}Skipped searching on indexers for ${searcheeLog}${reason} | MediaType: ${mediaTypeLog} | IDs: N/A`;
+
+	const skippingIndexersCallout =
+		enabledIndexers.length > indexersToUse.length
+			? [" (filtered by"]
+			: [""];
+	if (enabledIndexers.length > indexersToUseByTimestamp.length) {
+		skippingIndexersCallout.push(
+			indexersToUseByTimestamp.length === indexersToUse.length
+				? "timestamps)"
+				: "timestamps",
+		);
+	}
+	if (indexersToUseByTimestamp.length > indexersToUse.length) {
+		skippingIndexersCallout.push(
+			enabledIndexers.length === indexersToUseByTimestamp.length
+				? "supported category)"
+				: "and supported category)",
+		);
+	}
+	const reasonStr = new Intl.ListFormat("en", {
+		style: "narrow",
+		type: "unit",
+	}).format(skippingIndexersCallout);
+	if (indexersToUse.length === 0) {
+		logger.info(`${skippingSearching(reasonStr)}`);
+		return { indexersToUse };
```

Comment:
this is not really taking advantage of Intl.ListFormat, which if used idiomatically, will handle all of the things you're doing with ternaries.

Try this: 
```js
const filteringCauses = [
  areIndexersBeingFilteredByTimestamps,
  areIndexersBeingFilteredBySupportedCategory,
].filter(isTruthy);
const reasonStr = filteringCauses.length ? `(filtered by ${formatAsList(filteringCauses)})` ? '';
```